src/rete/controls/LabeledInputControl.tsx

```diff
@@ -1,5 +1,63 @@
 import { ClassicPreset as Classic } from 'rete'
 import { Presets } from 'rete-react-plugin'
+
+import * as React from 'react'
+import { ClassicPreset } from 'rete'
+import styled from 'styled-components'
+
+
+
+const Input = styled.input<{ styles?: (props: any) => any }>`
+  width: 100%;
+  border-radius: 30px;
+  background-color: white;
+  padding: 2px 6px;
+  border: 1px solid #999;
+  font-size: 110%;
+  box-sizing: border-box;
+  ${props => props.styles && props.styles(props)}
+`
+
+export class LabeledInputControl extends Classic.InputControl<'number'> {
+    constructor(
+        public value: number,
+        public label: string,
+        change?: () => void,
+        public increment?: number,
+        readonly?: boolean
+    ) {
+        super('number', { initial: value, readonly: Boolean(readonly), change })
+    }
+}
+export function CustomLabeledInputControl<N extends 'text' | 'number'>(props: { data: LabeledInputControl, styles?: () => any }) {
+  const [value, setValue] = React.useState(props.data.value)
+  const ref = React.useRef(null)
```

Comment:
The original source code has a Drag.useNoDrag(ref) in here - not entirely sure how it works, but it seems to be something we'd like to have too. If we don't have useNoDrag, we aren't able to use a drag/caret select on the control (because it'll drag the whole node instead of selecting the input text). Let's add that line back in and then functionality should be perfect!